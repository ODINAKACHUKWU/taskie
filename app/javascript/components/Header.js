import React from "react";
import PropTypes from "prop-types";

const defaultProps = {
  title: "Taskie",
};

function Header({ title = defaultProps.title, action }) {
  return (
    <div className="header-container">
      <div className="title">{title}</div>
      <div className="action-button">{action}</div>
    </div>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  action: PropTypes.element,
};

export default Header;
