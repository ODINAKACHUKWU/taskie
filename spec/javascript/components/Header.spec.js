import "jsdom-global/register";
import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Header from "../../../app/javascript/components/Header";

const defaultProps = {
  title: "Taskie",
  action: null,
};

describe("rendering Header component", () => {
  it("renders Header component without crashing", () => {
    shallow(<Header title={defaultProps.title} />);
  });

  it("renders title and action button", () => {
    const wrapper = mount(
      <Header title={defaultProps.title} action={defaultProps.action} />
    );
    const title = wrapper.find(".title").text();

    expect(wrapper.contains(title)).toEqual(true);
    expect(title).toEqual(defaultProps.title);
    expect(wrapper.exists(".action-button")).toEqual(true);
  });
});

describe("passing props", () => {
  it("accepts title and action props", () => {
    const wrapper = mount(
      <Header title={defaultProps.title} action={defaultProps.action} />
    );

    expect(wrapper.props().title).toEqual(defaultProps.title);
    expect(wrapper.props().action).toEqual(null);
  });
});

describe("snapshot", () => {
  it("Header snapshot", () => {
    const tree = shallow(<Header title={defaultProps.title} />);

    expect(toJSON(tree)).toMatchSnapshot();
  });
});
