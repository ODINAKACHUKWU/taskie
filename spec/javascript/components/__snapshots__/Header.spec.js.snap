// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Header snapshot 1`] = `
<div
  className="header-container"
>
  <div
    className="title"
  >
    Taskie
  </div>
  <div
    className="action-button"
  />
</div>
`;
